lib/medusa/index.ts

```diff
@@ -0,0 +1,337 @@
+import { isMedusaError } from 'lib/type-guards';
+
+import { mapOptionIds } from 'lib/utils';
+import {
+  Cart,
+  CartItem,
+  MedusaCart,
+  MedusaLineItem,
+  MedusaProduct,
+  MedusaProductCollection,
+  MedusaProductOption,
+  MedusaProductVariant,
+  Product,
+  ProductCollection,
+  ProductOption,
+  ProductVariant,
+  SelectedOption
+} from './types';
+
+// const endpoint = `${process.env.MEDUSA_BACKEND_API!}`;
+const endpoint = `http://localhost:9000/store`;
+
+export default async function medusaRequest(
+  method: string,
+  path = '',
+  payload?: Record<string, unknown> | undefined
+) {
+  const options: RequestInit = {
+    method,
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  };
+
+  if (payload) {
+    options.body = JSON.stringify(payload);
+  }
+
+  try {
+    const result = await fetch(`${endpoint}/${path}`, options);
+
+    const body = await result.json();
+
+    if (body.errors) {
+      throw body.errors[0];
+    }
+
+    return {
+      status: result.status,
+      body
+    };
+  } catch (e) {
+    if (isMedusaError(e)) {
+      throw {
+        status: e.status || 500,
+        message: e.message
+      };
+    }
+
+    throw {
+      error: e
+    };
+  }
+}
+
+const reshapeCart = (cart: MedusaCart): Cart => {
+  const lines = cart?.items?.map((item) => reshapeLineItem(item)) || [];
+  const totalQuantity = lines.length;
+  const checkoutUrl = '/';
+  const currencyCode = 'EUR';
+  const cost = {
+    subtotalAmount: {
+      amount:
```

Comment:
thought: Use `cart.subtotal` ?

Reply:
Haha, didn't realise that was available 😄 fixed it